test/runtests.jl

```diff
@@ -271,7 +275,7 @@ end
 
 @test_zip [1,2,3] [:a, :b, :c] ['x', 'y', 'z']
 @test_zip [1,2,3] [:a, :b] ['w', 'x', 'y', 'z']
-@test_zip [1,2,3] @compat(Union{})[] ['w', 'x', 'y', 'z']
```

Comment:
`[]` is `Union{}[]` on 0.3. And this feels like a corner case that still worth testing.